nessie/auth_test.go

```diff
@@ -0,0 +1,167 @@
+	require.NoError(t, err, "Developer unexpectedly failed to read branch commit")
+	require.Equal(t, http.StatusOK, resCom.StatusCode(), "Developer unexpectedly failed to read branch commit")
+
+	// creating a branch should be authorized
+	branch1 := "feature-1"
+	resAddBch, err := developerClient.CreateBranchWithResponse(ctx, repo, api.CreateBranchJSONRequestBody{
+		Name:   branch1,
+		Source: mainBranch,
+	})
+	require.NoError(t, err, "Developer failed while testing create branch")
+	require.Equal(t, http.StatusCreated, resAddBch.StatusCode(), "Developer unexpectedly failed to create branch")
+
+	// attempting to delete the repository should be authorized
+	resDelRep, err := developerClient.DeleteRepositoryWithResponse(ctx, repo)
+	require.NoError(t, err, "Developer failed while testing delete repository")
+	require.Equal(t, http.StatusUnauthorized, resDelRep.StatusCode(), "Developer unexpectedly did not receive unauthorized response while deleting repo")
+}
+
+// Test Super User Policies: AuthManageOwnCredentials, FSFullAccess, RepoManagementReadAll
+func TestSuperUserPolicies(t *testing.T) {
+	ctx, _, repo := setupTest(t)
+
+	gid := "SuperUsers"
+	endpointURL := viper.GetString("endpoint_url") // defined in setup.go
+
+	// generate the SuperUser client
+	superUserClient, err := NewClientFromGroup(t, ctx, gid, endpointURL)
+	require.NoError(t, err, "failed to initialize client with group")
+
+	// listing the available braches should succeed
+	resBch, err := superUserClient.ListBranchesWithResponse(ctx, repo, &api.ListBranchesParams{})
+	require.NoError(t, err, "SuperUser unexpectedly failed to list branches of repository")
+	require.Equal(t, http.StatusOK, resBch.StatusCode(), "SuperUser unexpectedly failed to list branches of repository")
+
+	branches := resBch.JSON200.Results
+
+	// reading the commit of the main branch of the repo should succeed
+	resCom, err := superUserClient.GetCommitWithResponse(ctx, repo, branches[0].CommitId)
+	require.NoError(t, err, "SuperUser unexpectedly failed to read branch commit")
+	require.Equal(t, http.StatusOK, resCom.StatusCode(), "SuperUser unexpectedly failed to read branch commit")
+
+	// creating a branch should succeed
+	branch1 := "feature-1"
+	resAddBch, err := superUserClient.CreateBranchWithResponse(ctx, repo, api.CreateBranchJSONRequestBody{
+		Name:   branch1,
+		Source: mainBranch,
+	})
+	require.NoError(t, err, "SuperUser failed while testing create branch")
+	require.Equal(t, http.StatusCreated, resAddBch.StatusCode(), "SuperUser unexpectedly failed to create branch")
+
+	// attempting to delete the repository should be authorized and result in no content response
+	resDelRep, err := superUserClient.DeleteRepositoryWithResponse(ctx, repo)
+	require.NoError(t, err, "SuperUser failed while testing delete repository")
+	require.Equal(t, http.StatusNoContent, resDelRep.StatusCode(), "SuperUser unexpectedly did not receive no content response while deleting repo")
```

Comment:
The admin can do so much more, like managing groups. Let's add more checks for those as well.